Detections/MultipleDataSources/MalformedUserAgents.yaml

```diff
@@ -30,13 +33,23 @@ query: |
   (OfficeActivity | where TimeGenerated >= ago(endtime) | where UserAgent != ""),
   (OfficeActivity
   | where TimeGenerated >= ago(endtime)
-  | where RecordType in ("AzureActiveDirectoryAccountLogon", "AzureActiveDirectoryStsLogon")
-  | extend OperationName = Operation, UserAgent = extractjson("$[0].Value", ExtendedProperties, typeof(string))),
+  | where RecordType in ("AzureActiveDirectory", "AzureActiveDirectoryStsLogon")
+  | extend OperationName = Operation
+  | parse ExtendedProperties with * 'User-Agent\\":\\"' UserAgent2 '\\' *
+  | parse ExtendedProperties with * 'UserAgent",      "Value": "' UserAgent1 '"' *
+  | where isnotempty(UserAgent1) or isnotempty(UserAgent2)
+  | extend UserAgent = iff( RecordType == 'AzureActiveDirectoryStsLogon', UserAgent1, UserAgent2)
+  ),
   (AzureDiagnostics
   | where TimeGenerated >= ago(endtime)
   | where ResourceType =~ "APPLICATIONGATEWAYS" 
   | where OperationName =~ "ApplicationGatewayAccess" 
-  | extend ClientIP = columnifexists("clientIP_s", "None"), UserAgent = columnifexists("userAgent_s", "None")))
+  | extend ClientIP = columnifexists("clientIP_s", "None"), UserAgent = columnifexists("userAgent_s", "None")
+  ),
+  (SigninLogs
+  | where TimeGenerated >= ago(endtime)
+  | extend ClientIP = IPAddress )
+  )
   // Likely artefact of hardcoding
```

Comment:
Look at adding in AADNonInteractiveUserSignInLogs and it might make sense to join TI logs based on the UserAgent so we have a row that indicates a TI connection where applicable.

Reply:
Added AADNonInteractiveUserSignInLogs. Adding TI Logs with join was getting interesting results. Will probably investigate in later iteration.

Reply:
Thanks Shain.